### DISCUSSION TITLE
Larger image preview

### DISCUSSION THREAD

**Discussion by Traslogan** [vladmandic/sdnext#753] [Ideas]
Hi, I've been using Vladmandic to get around some of the issues of Auto1111, it's great but it feels like the image previews in the text-to-image and whatnot are rather small. Could we have the option to make the previews bigger in some way? (It becomes especially small previews if the image being generated is tall)

**Comment by vladmandic** [comment]
you mean in the full-screen preview when you click on image or in the gallery view on the right side?

**Reply by Traslogan** [reply]
Before you click on anything, the standard preview. It's very small and leaves a lot of negative space under it, it would be nice to be able resize it so the standard preview is a bit bigger.

**Reply by vladmandic** [reply]
size of thumbnail in gallery view is now determined based number of columns displayed.
and number of columns is determined based on screen size.

those are settings i'm using: `columns=[1,2,3,4,5,6])`
and they refer to screen sizes <576px, <768px, <992px, <1200px, <1400px, >1400px

**Comment by ghost** [comment]
I am curious how to increase the size of the live preview window. I have tried editing the config.json and nothing changes.

**Reply by vladmandic** [reply]
you mean how to reduce number of columns in gallery view so each live preview appears larger? you cant at the moment. but if would be a valid feature request if someone created it.

**Reply by ghost** [reply]
No when rendering a new image the live preview shows a single image. What I want to accomplish is making the window for the live preview larger so the preview image is also larger.

I am not referring to the image group where it shows multiple renders in a grid. I am referring to a single image render, and not the finished render preview either. Live preview while rendering is what I want larger.

**Comment by vladmandic** [comment]
you can create `user.css` to override default style, for example

```css
#txt2img_gallery { height: 4000px; }
#txt2img_gallery img { object-fit: cover; }
```

**Reply by tazztone** [reply]
thanks works great! but still having the issue that the dark box is not expanding with the gallery and because of that a square image is cut off at the top/botom
![2023-06-24 11 34 52](https://github.com/vladmandic/automatic/assets/62671577/7f322633-1be4-4e7a-80c2-2ee869518b7b)
any fix for that? couldn't find out code via "inspect"

**Comment by ghost** [comment]
> you can create `user.css` to override default style, for example
> 
> ```css
> #txt2img_gallery { height: 4000px; }
> #txt2img_gallery img { object-fit: cover; }
> ```

That worked perfectly! Wow this awesome! Thank you!~
I have my height set to around 600-800px and it looks much better! Thank you soo much!~

**Comment by Tillerz** [comment]
Is there a way to move the buttons from below the preview window to above? I like to Generate + Save a lot. And if the window for previews is larger, then need to scroll up/down every time. :)

**Comment by vladmandic** [comment]
css tricks :)

```css
#txt2img_gallery_container { order: 1; }
```

**Reply by Tillerz** [reply]
That did work. :) But it also moved all the generation metadata up so it's pushing the image down where it's half invisible. That did't really improve things. ;-)
